### PR TITLE
[TEP-0138] Decouple API and Feature Versioning Problem Statement

### PR DESCRIPTION
**Pull request by JeromeJu**
This commit adds the problem statement for decoupling api and feature versioning. This TEP is marked as 'proposed'.

### CODE REVIEW

**Review by QuanZhang-William — COMMENTED** [review]
Thanks Jerome for putting everything together.

nit: could you please make the `v1`/ `v1beta1` formatting consistent in the doc (i.e. v1 -> `v1`)

File: teps/0138-decouple-api-and-feature-versioning.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+- When promoting features, it could cause confusions for contributors to be dependent on the fact whether an apiVersion is available. For example, during [the promotion to beta for projected workspaces](https://github.com/tektoncd/pipeline/pull/5530), the `v1` api's existence led to confusions of what to do with `beta` features in `v1beta1` and its difference with in `v1`.
+
+### Goals
+
+- Feature validation and implementation should be independant from any API version.
+- Come up with a backward-compatible migration plan for setting the `enable-api-fields` feature flag to `stable` in the long term.
+- Changes and updates made to the existent feature validaitons regarding decoupling api and feature versioning should keep much backwards compatiblity as possible.
```
Comment (vdemeester) [inline teps/0138-decouple-api-and-feature-versioning.md:41]: `s/validaitons/validations/`

File: teps/0138-decouple-api-and-feature-versioning.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+**End Users**
+- As a users who has newly adopted Tekton:
+  - I want to have a consistent and easily understandable feature flag UX.
+
+- As an end user currently on `v1beta1`:
+  - I want to migrate to `v1` and have as seamless of an experience as possible.
+
+**Cluster Operators**
+- As an cluster operator whose most users are on `v1beta1`:
```
Comment (vdemeester) [inline teps/0138-decouple-api-and-feature-versioning.md:61]: "As a cluster operator …"

File: teps/0138-decouple-api-and-feature-versioning.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  - I want to control the features that my users use and have enough notice of any backwards incompatible changes going to be made in the Tekton pipeline releases.
+  - When migrating to `v1`, I may want my users to keep using `stable` opt-in features that have already been turned on by default in `v1beta1`.
+
+- As a Cluster operator with users who have migrated to `v1`:
+  - I would like to get notice of the plan for the breaking change if `enable-api-fields` is going to be changed to `stable` in the future.
+
+- As a clsuter operator who accept default values of all pipeline releases:
```
Comment (vdemeester) [inline teps/0138-decouple-api-and-feature-versioning.md:68]: `s/clsuter/cluster`

**Review by vdemeester — COMMENTED** [review]
Few typos, otherwise LGTM for the problem statement.

File: teps/0138-decouple-api-and-feature-versioning.md (hunk 1 above)
Comment (JeromeJu) [reply teps/0138-decouple-api-and-feature-versioning.md:41]: Thanks Vincent!

**Pull request merged**